nx/test/nx/lin_alg_test.exs

```diff
@@ -63,65 +63,91 @@ defmodule Nx.LinAlgTest do
 
-      assert round(q, 1) ==
-               Nx.tensor([
-                 [2 / 3, 2 / 3, 1 / 3],
-                 [2 / 3, -1 / 3, -2 / 3],
-                 [1 / 3, -2 / 3, 2 / 3]
-               ])
-               |> round(1)
+      expected_q =
+        Nx.tensor([
+          [2 / 3, 2 / 3, 1 / 3],
+          [2 / 3, -1 / 3, -2 / 3],
+          [1 / 3, -2 / 3, 2 / 3]
+        ])
 
-      assert round(r, 1) ==
-               Nx.tensor([
-                 [3.0, 0.0, 12.0],
-                 [0.0, -3.0, 12.0],
-                 [0.0, 0.0, 6.0]
-               ])
-               |> round(1)
+      assert_all_close(q, expected_q, rtol: 0, atol: 0)
+
+      expected_r =
+        Nx.tensor([
+          [3.0, 0.0, 12.0],
+          [0.0, -3.0, 12.0],
+          [0.0, 0.0, 6.0]
+        ])
+
+      assert_all_close(r, expected_r, rtol: 0, atol: 0)
+
+      assert_all_close(Nx.dot(q, r), t, rtol: 0, atol: 0)
     end
 
     test "factors rectangular matrix" do
       t = Nx.tensor([[1.0, -1.0, 4.0], [1.0, 4.0, -2.0], [1.0, 4.0, 2.0], [1.0, -1.0, 0.0]])
+
+      # Reduced mode
       {q, r} = Nx.LinAlg.qr(t, mode: :reduced)
 
-      assert round(q, 1) ==
-               Nx.tensor([
-                 [0.5774, -0.8165, 0.0],
-                 [0.5774, 0.4082, -0.7071],
-                 [0.5774, 0.4082, 0.7071],
-                 [0.0, 0.0, 0.0]
-               ])
-               |> round(1)
+      expected_q =
+        Nx.tensor([
+          [0.5, -0.5, 0.5],
+          [0.5, 0.5, -0.5],
+          [0.5, 0.5, 0.5],
+          [0.5, -0.5, -0.5]
+        ])
 
-      assert round(r, 1) ==
-               Nx.tensor([
-                 [1.7321, 4.0415, 2.3094],
-                 [0.0, 4.0825, -3.266],
-                 [0.0, 0.0, 2.8284]
-               ])
-               |> round(1)
+      assert_all_close(q, expected_q, rtol: 0, atol: 0)
 
-      assert Nx.tensor([
-               [1.0, -1.0, 4.0],
-               [1.0, 4.0, -2.0],
-               [1.0, 4.0, 2.0],
-               [0.0, 0.0, 0.0]
-             ]) == q |> Nx.dot(r) |> round(1)
+      expected_r =
+        Nx.tensor([
+          [2.0, 3.0, 2.0],
+          [0.0, 5.0, -2.0],
+          [0.0, 0.0, 4.0]
+        ])
+
+      assert_all_close(r, expected_r, rtol: 0, atol: 0)
+
+      assert_all_close(Nx.dot(q, r), t, rtol: 0, atol: 0)
+
+      # Complete mode
+      {q, r} = Nx.LinAlg.qr(t, mode: :complete)
+
+      expected_q =
+        Nx.tensor([
+          [0.5, -0.5, 0.5, -0.5],
+          [0.5, 0.5, -0.5, -0.5],
+          [0.5, 0.5, 0.5, 0.5],
+          [0.5, -0.5, -0.5, 0.5]
+        ])
+
+      assert_all_close(q, expected_q, rtol: 0, atol: 0)
+
+      expected_r =
+        Nx.tensor([
+          [2.0, 3.0, 2.0],
+          [0.0, 5.0, -2.0],
+          [0.0, 0.0, 4.0],
+          [0.0, 0.0, 0.0]
+        ])
+
+      assert_all_close(r, expected_r, rtol: 0, atol: 0)
+
+      assert_all_close(Nx.dot(q, r), t, rtol: 0, atol: 0)
```

Comment:
bold! haha

Reply:
I relaxed the condition because the exact match may be too strict depending on environments.